homeassistant/components/overkiz/select.py

```diff
@@ -0,0 +1,134 @@
+        }[OverkizCommandParam(option)],
+        None,
+    )
+
+
+def _select_option_memorized_simple_volume(
+    option: str, execute_command: Callable[..., Awaitable[None]]
+) -> Awaitable[None]:
+    """Change the selected option for Memorized Simple Volume."""
+    return execute_command(OverkizCommand.SET_MEMORIZED_SIMPLE_VOLUME, option)
+
+
+SELECT_DESCRIPTIONS: list[OverkizSelectDescription] = [
+    OverkizSelectDescription(
+        key=OverkizState.CORE_OPEN_CLOSED_PEDESTRIAN,
+        name="Position",
+        icon="mdi:content-save-cog",
+        options=[
+            OverkizCommandParam.OPEN,
+            OverkizCommandParam.PEDESTRIAN,
+            OverkizCommandParam.CLOSED,
+        ],
+        select_option=_select_option_open_closed_pedestrian,
+        device_class=OverkizDeviceClass.OPEN_CLOSED_PEDESTRIAN,
+    ),
+    OverkizSelectDescription(
+        key=OverkizState.IO_MEMORIZED_SIMPLE_VOLUME,
+        name="Memorized Simple Volume",
+        icon="mdi:volume-high",
+        options=[OverkizCommandParam.STANDARD, OverkizCommandParam.HIGHEST],
+        select_option=_select_option_memorized_simple_volume,
+        entity_category=EntityCategory.CONFIG,
+        device_class=OverkizDeviceClass.MEMORIZED_SIMPLE_VOLUME,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz select from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizSelect] = []
+
+    key_supported_states = {
+        description.key: description for description in SELECT_DESCRIPTIONS
+    }
+
+    for device in data.coordinator.data.values():
+        if (
+            device.widget in IGNORED_OVERKIZ_DEVICES
+            or device.ui_class in IGNORED_OVERKIZ_DEVICES
+        ):
+            continue
+
+        for state in device.definition.states:
+            if description := key_supported_states.get(state.qualified_name):
+                entities.append(
+                    OverkizSelect(
+                        device.device_url,
+                        data.coordinator,
+                        description,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class OverkizSelect(OverkizDescriptiveEntity, SelectEntity):
+    """Representation of an Overkiz Select entity."""
+
+    entity_description: OverkizSelectDescription
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        if state := self.device.states.get(self.entity_description.key):
+            return str(state.value)
+
+        return None
+
+    @property
+    def options(self) -> list[str]:
+        """Return a set of selectable options."""
+        return self.entity_description.options
```

Comment:
I was a bit surprised that `SelectEntityDescription` didn't support this natively :(